core/src/consensus.rs

```diff
@@ -520,9 +530,16 @@ impl Tower {
                             // all of them.
                             panic!("no ancestors found with slot: {}", last_voted_slot);
                         } else {
-                            // bank_forks doesn't have corresponding data for the stray restored last vote,
-                            // meaning some inconsistency between saved tower and ledger.
-                            // (newer snapshot, or only a saved tower is moved over to new setup?)
+                            // compare slots not to error! just because of newer snapshots
+                            if self.is_first_switch_check() && switch_slot < last_voted_slot {
+                                error!(
```

Comment:
The reason I didn't `error!` here at #10718 is that doing so printed so many duplicate errors because `check_switch_threshold` is called frequently. Thanks to `is_first_switch_check()` (and `last_switch_threshold_check` behind it) this isn't true anymore.